src/foolscap/banana.py

```diff
@@ -10,31 +11,47 @@
 from foolscap.slicers.allslicers import RootSlicer, RootUnslicer
 from foolscap.slicers.allslicers import ReplaceVocabSlicer, AddVocabSlicer
 
-import stringchain
-import tokens
-from tokens import SIZE_LIMIT, STRING, LIST, INT, NEG, \
-     LONGINT, LONGNEG, VOCAB, FLOAT, OPEN, CLOSE, ABORT, ERROR, \
-     PING, PONG, \
-     BananaError, BananaFailure, Violation
+from . import stringchain
+from . import tokens
+from .tokens import (SIZE_LIMIT, STRING, LIST, INT, NEG, 
+                    LONGINT, LONGNEG, VOCAB, FLOAT, OPEN, CLOSE, ABORT, ERROR, 
+                    PING, PONG, 
+                    BananaError, BananaFailure, Violation)
 
```

Comment:
> Okay, after churning through this monster I'd like to re-iterate my previous general comment: this is really too big and too many different kinds of changes to review properly. It would be way better if, e.g., you had a single PR for "failUnless -> assert.." changes, one for "six.bytes" related changes, etc etc. because then we can merge some things faster.
> 
> As it stands, this needs changes in some spots .. but I'll have to re-read through all the "simple/mechanical" changes again every time ... :/

I understand the spirit of this but for a port and to keep the tests passing with the changes you understand what you are saying is difficult. 

I can rework this to multiple changes  like you requested from next PR - so pls confirm if you want the current PR to be reworked or if its fine from the next one. Thanks.

Reply:
> I don't know what the convention (if any?) in foolscap is, but in Tahoe we'd typically indent 
> follow-on lines at 4 spaces.
Personally I tend to align the starting character of each module to same column, so that is what I have done here. And I use tab-to-spaces in my Emacs so any final code will have tabs converted to spaces. Just saying. Thanks.

Reply:
> I can rework this to multiple changes like you requested from next PR - so pls confirm if you want the current PR to be reworked or if its fine from the next one. Thanks.

I think there are lots of changes in here that would keep tests passing .. e.g. any whitespace changes, any of the test-assertion-function changes, all the `print` -> `print()` changes, using `six.bytes` (or `b""` / `u""`) etc. I guess it's up to you for this PR, but those would make some fine "smaller" PRs and could be easily merged.

Reply:
> For imports, I tend to align the starting character of each module to same column for multi-line imports,

Okay, but it seems the convention is to use 4-space indenting already on continuation lines in Foolscap?

Reply:
Ok. I can rework this into multiple PRs. Let me do that. First one would be the one with Zope changes.

Reply:
See https://github.com/warner/foolscap/pull/53 . Thanks.